### ISSUE TITLE
[JENKINS-28595] Tests execution stops in a deadlock

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#659]
<p>When I try to run the tests the result is a hanged process .</p>

<p>It seems to always stop in this test:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Running hudson.tools.ToolLocationTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.004 sec - in hudson.tools.ToolLocationTest
</pre>
</div></div>

<p>You find attached information related to the process using jstack (output.txt)</p>

<p>This <a href="https://jenkins.ci.cloudbees.com/job/core/job/jenkins-core/2677/" class="external-link" target="_blank" rel="nofollow noopener">PR</a> can not be merged because it is blocked by this issue. Its execution ends with a timeout.</p>

**Attachments:**

- [output.txt](https://issues.jenkins.io/secure/attachment/29851/output.txt)

---
<details><summary><i>Originally reported by <a title="Manuel Recena Soto" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena">recena</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-28595" target="_blank">Tests execution stops in a deadlock</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a title="Manuel Recena Soto" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=recena">recena</a>
<li><b>reported by</b>: recena
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2015-11-15T15:43:56+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>:</i>
<p>AFAIK Manuel is working on the issue</p>
